### ISSUE TITLE
IllegalStateException: Can not perform this action after onSaveInstanceState

### ISSUE THREAD

**Issue by kspearrin** [smstuebe/xamarin-fingerprint#92]
## Steps to reproduce

1. Use bitwarden app with Fingerprint enabled.

2. Observe crash occurring to certain users when fingerprint unlocking presented.

## Expected behavior
No crash.

## Actual behavior
I am seeing this exception in our HockeyApp crash logs a lot as of late. It doesn't seem to be widespread since I cannot reproduce it, but it is happening quite often for certain users.

Implementation: https://github.com/bitwarden/mobile/blob/master/src/App/Pages/Lock/LockFingerprintPage.cs#L101

### Crashlog

```
java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
	at android.app.FragmentManagerImpl.checkStateLoss(FragmentManager.java:1438)
	at android.app.FragmentManagerImpl.enqueueAction(FragmentManager.java:1456)
	at android.app.BackStackRecord.commitInternal(BackStackRecord.java:707)
	at android.app.BackStackRecord.commit(BackStackRecord.java:671)
	at android.app.DialogFragment.show(DialogFragment.java:233)
	at mono.java.lang.RunnableImplementor.n_run(Native Method)
	at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6692)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
Xamarin caused by: Java.Lang.IllegalStateException: Can not perform this action after onSaveInstanceState
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at Java.Interop.JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00089] in <d24d79ea6b2c4db5a24d5f19693dbc99>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0005d] in <d24d79ea6b2c4db5a24d5f19693dbc99>:0 
  at Android.App.DialogFragment.Show (Android.App.FragmentManager manager, System.String tag) [0x0004b] in <af337c069bfa4c42b8fee92a424b3859>:0 
  at Plugin.Fingerprint.Dialog.FingerprintDialogFragment.Show (Android.App.FragmentManager manager, System.String tag) [0x0000b] in <ea79f217bb7b4de0af17f31402ea8d2c>:0 
  at Plugin.Fingerprint.Dialog.FingerprintDialogFragment+<ShowAsync>d__20.MoveNext () [0x00036] in <ea79f217bb7b4de0af17f31402ea8d2c>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at Plugin.Fingerprint.Contract.AndroidFingerprintImplementationBase+<NativeAuthenticateAsync>d__0.MoveNext () [0x0008a] in <ea79f217bb7b4de0af17f31402ea8d2c>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at Plugin.Fingerprint.Abstractions.FingerprintImplementationBase+<AuthenticateAsync>d__1.MoveNext () [0x000f2] in <d24d4fa4f12d44f5828af948aec88dfd>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at Bit.App.Pages.LockFingerprintPage+<CheckFingerprintAsync>d__9.MoveNext () [0x000e5] in <f0675b5e4e2448f595a04aa556909ab1>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at Bit.App.Pages.LockFingerprintPage+<OnAppearing>d__8.MoveNext () [0x000f1] in <f0675b5e4e2448f595a04aa556909ab1>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in <dbebbc9d923441e79a2a2a39fb18cd51>:0 
  at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <af337c069bfa4c42b8fee92a424b3859>:0 
  at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <af337c069bfa4c42b8fee92a424b3859>:0 
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in <af337c069bfa4c42b8fee92a424b3859>:0 
  at (wrapper dynamic-method) System.Object:7b2dcba9-5aa0-4f5c-9613-ba6f5248ada1 (intptr,intptr)
  --- End of managed Java.Lang.IllegalStateException stack trace ---
java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
	at android.app.FragmentManagerImpl.checkStateLoss(FragmentManager.java:1438)
	at android.app.FragmentManagerImpl.enqueueAction(FragmentManager.java:1456)
	at android.app.BackStackRecord.commitInternal(BackStackRecord.java:707)
	at android.app.BackStackRecord.commit(BackStackRecord.java:671)
	at android.app.DialogFragment.show(DialogFragment.java:233)
	at mono.java.lang.RunnableImplementor.n_run(Native Method)
	at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6692)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
```

## Configuration

**Version of the Plugin:** 1.4.6-beta4

**Platform:** Android 7.0, 7.1.1, 8.0

**Device:** SM-G935F (7.0), Moto Z2 Play (7.1.1), ONEPLUS A500 (8.0), ONEPLUS A3003 (8.0)

**Comment by smsissuechecker** [comment]
Hi @kspearrin,

I'm the friendly issue checker.
Thanks for using the issue template :star2:
I appreciate it very much. I'm sure, the maintainers of this repository will answer, soon.

**Comment by SamuelDebruyn** [comment]
Experienced the same issue. I see crashes in App Center. I think this happens when the app is sent to background while the user is authenticating. This is one of the stack traces:

```
FingerprintDialogFragment+<Dismiss>d__23.MoveNext ()
Java.Lang.IllegalStateException: Can not perform this action after onSaveInstanceState
ExceptionDispatchInfo.Throw ()
JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args)
JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters)
DialogFragment.Dismiss ()
FingerprintDialogFragment+<Dismiss>d__23.MoveNext ()
ExceptionDispatchInfo.Throw ()
AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state)
SyncContext+<>c__DisplayClass2_0.<Post>b__0 ()
Thread+RunnableImplementor.Run ()
IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this)
(wrapper dynamic-method) System.Object:4500e374-54c6-4dce-9721-8a2d6d6d2b0f (intptr,intptr)
java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
android.app.FragmentManagerImpl.checkStateLoss(FragmentManager.java:1436)
android.app.FragmentManagerImpl.enqueueAction(FragmentManager.java:1454)
android.app.BackStackRecord.commitInternal(BackStackRecord.java:707)
android.app.BackStackRecord.commit(BackStackRecord.java:671)
android.app.DialogFragment.dismissInternal(DialogFragment.java:301)
android.app.DialogFragment.dismiss(DialogFragment.java:267)
md5cd42f6d72bf05f50197800f6ea586bee.TaskAnimationListener.n_onAnimationEnd(Native Method)
md5cd42f6d72bf05f50197800f6ea586bee.TaskAnimationListener.onAnimationEnd(TaskAnimationListener.java:41)
android.animation.ValueAnimator.endAnimation(ValueAnimator.java:1149)
android.animation.ValueAnimator.doAnimationFrame(ValueAnimator.java:1309)
android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)
android.animation.AnimationHandler.-wrap2(AnimationHandler.java)
android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)
android.view.Choreographer$CallbackRecord.run(Choreographer.java:869)
android.view.Choreographer.doCallbacks(Choreographer.java:683)
android.view.Choreographer.doFrame(Choreographer.java:616)
android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:857)
android.os.Handler.handleCallback(Handler.java:751)
android.os.Handler.dispatchMessage(Handler.java:95)
android.os.Looper.loop(Looper.java:154)
android.app.ActivityThread.main(ActivityThread.java:6123)
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
```

**Comment by kspearrin** [comment]
This is what my HockeyApp crash reports are looking like as of late

![image](https://user-images.githubusercontent.com/1190944/38044303-55e278f6-3287-11e8-8411-f6c37344128c.png)

**Comment by smstuebe** [comment]
meh. I hope all that stuff will be resolved with Android P (see #97). Not sure if it's worth the effort. What do you think?
41 users. How many % of your users are that?

**Comment by kspearrin** [comment]
It's more than 41 from just the first line, but all in it's a pretty low percentage of users. It's more of just an annoyance cluttering up my crash logs.

**Comment by jaspervanmegroot** [comment]
Have the same issue here

**Comment by ivanicin** [comment]
As far as my users observed, this is actually happening when you:
1. use fingerprint and unlock the app
2. move the app to the background

> meh. I hope all that stuff will be resolved with Android P (see #97). Not sure if it's worth the effort. What do you think?
> 41 users. How many % of your users are that?
As far as Android adaption goes, it will be like 4 years from now before you can actually rely only on Android P API. What do you think?
